app/models/email_configuration.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class EmailConfiguration < ApplicationRecord
+
+  validates :email, :password, :user_name, :authentication, :domain, :tls, :address, length: { maximum: 255 }
+
+  validates :email, :password, :user_name, presence: true
+  validates :email, format: { with: Authentication::RE_EMAIL_OK, message: Authentication::MSG_EMAIL_BAD }
+  validates :email, uniqueness: { case_sensitive: false, message: "This email address is already used." }
+  validates :address, :domain, format: { with: Authentication::RE_DOMAIN_OK }, if: -> { address }
+  validates :authentication, inclusion: { in: %w[plain login cram_md5] }, if: -> { authentication }
+  validates :tls, inclusion: { in: %w[starttls starttls_auto tls no] }, if: -> { tls }
```

Comment:
I would not do that in this first PR but in a second one. Not even displaying the field